### PR TITLE
feat(2423): implement password reset

### LINKED ISSUE #2423: Webpage Integration: Password Reset
## Goal
 
Wire up the full password reset flow to the backend, implement backend password validation, and create the password reset email template.
 
[Figma](https://www.figma.com/design/5j0fQssrV9ipoU16P7hfKy/Website-Deliverables?node-id=5760-24728&t=vZWJaoznUsHhdFqw-4)

---
 
## Scope
 
Full-stack: reset request, email delivery, token validation, password update, and confirmation email template.
 
---
 
### Step 1 — Request password reset
 
- The user submits their email address to request a password reset.
- If the email is registered, a reset email is sent to that address.
- If the email is not registered, an email is sent to the address informing the user that they are not registered and gives them a path to register; this email exists.
### Step 2 — Password reset email
 
- A password reset email template is created as part of this ticket.
- The email includes a reset link containing a time-limited, single-use token.
> **Email copy placeholder:**
> Subject: [TBD]
> Body: [Email copy will be added here]

[Email Template Design](https://www.figma.com/design/5j0fQssrV9ipoU16P7hfKy/Website-Deliverables?node-id=6635-16411&t=vZWJaoznUsHhdFqw-4)

### Step 3 — Reset link validation
 
- When the user clicks the reset link, the token is validated.
- If the token is valid, the user is taken to the new password form.
- If the token is expired or invalid, an error state is shown with a prompt to request a new reset link.
### Step 4 — Set new password
 
- The user enters and confirms a new password.
- Frontend validation is covered in a separate ticket.
- Backend validation is triggered only on form submission (i.e. clicking the CTA).
  - **Password not commonly used:** If the submitted password appears in a common passwords list, return an error and display an inline message: "This password is too common. Please choose a different one."
### Step 5 — Confirmation
 
- On successful password reset, the user is shown a confirmation state and redirected to the sign in page.
---
 
## FE/BE Actions
 
| Description | URL |
|---|---|
| Submit password reset request |   |
| Validate reset token |  |
| Submit new password |  |
 
---
 
## Out of Scope
 
- Frontend field validation (covered in a separate ticket)
- Sign in page

### PR DESCRIPTION
**Pull request by ycanales**
# Issue: #2423
## Summary & Context
Implements the password reset flow.

- **Figma link**: https://www.figma.com/design/5j0fQssrV9ipoU16P7hfKy/Website-Deliverables?node-id=5760-24728&t=vZWJaoznUsHhdFqw-4
- **Link to page:** http://localhost:8000/accounts/password/reset/

## Changes

- Adds email templates.
- Adds `AccountAdapter` 
- Password reset view is now a subclass of PasswordResetView from allauth

## ‼️ Risks & Considerations ‼️
*Please list any potential risks or areas that need extra attention during review/testing*
- ~`ACCOUNT_EMAIL_SUBJECT_PREFIX = ""` could it affect existing emails?~
    - Example of subject in production: _[www.boost.org] Unknown Account_
    - Subject prefix restored.
- The custom ACCOUNT_ADAPTER intercepts all of allauth mail, are we modifying existing ones?

## Screenshots
Complete flow:

Non registered email:

image

## Self-review Checklist
- [x] Link this PR to the related GitHub Project ticket

### Frontend
- [x] UI implementation matches Figma design
- [x] Tested in light and dark mode
- [x] Responsive / mobile verified
- [x] Accessibility checked (keyboard navigation, etc.)
- [x] Ensure design tokens are used for colors, spacing, typography, etc. – No hardcoded values
- [x] Test without JavaScript (if applicable)
- [x] No console errors or warnings

## Summary by CodeRabbit

* **New Features**
  * Added a branded “unknown account” password reset email with a signup-oriented call to action.
  * Improved the V3 password reset experience with an “Invalid reset link” state and a “Request a new reset link” option.

* **Bug Fixes**
  * Made “common password” errors more actionable and ensured user-entered email is preserved on errors.
  * Improved reset form rendering by ensuring the email field reflects the submitted value.

* **Style**
  * Standardized email subjects and reused shared warning note wording for consistent branding.

* **Tests**
  * Added end-to-end coverage for the V3 password reset flow and related edge cases.

### CODE REVIEW

**Review by javiercoronadonarvaez — CHANGES_REQUESTED** [review]
Hey @ycanales this is great work! Overall it works pretty well and I'm not sure if this is part of your ticket's scope, but immediately after resetting the password, I'm not able to log in. I need to go back to homepage in order to successfully log in, but the actual password resetting functionality works appropriately.

https://github.com/user-attachments/assets/9d6ece0f-5de5-48f1-883b-432b33b65269

Let me know what you think.

**Comment by ycanales** [comment]
@javiercoronadonarvaez  Thanks so much for testing this, I'll have a look at the issue you experienced and see if it falls within scope :)

**Comment by ycanales** [comment]
I believe it would probably be addressed in a ticket to wiring up the V3 login page, I'll confirm now in the daily.

**Comment by ycanales** [comment]
@javiercoronadonarvaez per discussion in the daily, I've switched the login view to the non-v3, so now a complete flow can be done.

File: templates/emails/unknown_account.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{{ action_url }}
+
+Didn't ask for this? Maybe someone typed your email by accident. You can
+ignore this email safely.
+
+--
+Boost.org is supported by grants from The C++ Alliance.
+
+Want to change how you receive these emails?
+You can update your preferences or unsubscribe from this list.
```
[the remark this answers is not in the document]
Comment (ycanales) [reply templates/emails/unknown_account.txt]: If the user received this because he doesn't have an account, it doesn't make sense to include that footer. Adjusted.

File: config/settings.py [hunk 2]
```diff
@@ -416,6 +416,9 @@
 
 # Django Allauth settings
 
+ACCOUNT_ADAPTER = "users.adapters.AccountAdapter"
+# Subjects come fully formed from the branded email templates
+ACCOUNT_EMAIL_SUBJECT_PREFIX = ""
```
Comment (julhoang) [inline config/settings.py]: Per our Slack discussion [here](https://metalab.slack.com/archives/C0ABKT42JVC/p1782756705878619), would you mind removing this `""` so that the subject lines can include the Boost domain?

**Review by julhoang — COMMENTED** [review]
Hi @ycanales ! While testing this locally, I noticed that the `PASSWORD_RESET_TIMEOUT` / `password_reset_link_lifetime` variable in the email doesn't seem to get set for me. I'm a little confused since I thought `PASSWORD_RESET_TIMEOUT` is a Django setting that has a default of 3 days. Do you happen to know why this is happening? 🤔

Image

File: config/settings.py (hunk 2 above)
Comment (ycanales) [reply config/settings.py]: Good catch, thanks @julhoang ! I've addressed both:

image

**Review by julhoang — APPROVED** [review]
Looks great to me, thank you for the updates @ycanales ! 🙌

**Review by jlchilders11 — APPROVED** [review]
Looks good to me, thank you for all the work!

File: users/views.py [hunk 3]
```diff
@@ -644,6 +663,8 @@ def get_v3_context_data(self, **kwargs):
             "img/v3/auth-page/auth-page-background.png"
         )
         context["login_url"] = reverse_lazy("v3-login")
+        # TODO: Switch to "v3-login" when it's ready.
+        context["login_url"] = reverse_lazy("account_login")
```
Comment (herzog0) [inline users/views.py]: isn't the second one overriding the first one above? which should we keep for now?

File: users/validators.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from django.contrib.auth.password_validation import (
+    CommonPasswordValidator as DjangoCommonPasswordValidator,
+)
+from django.utils.translation import gettext as _
+
+
+class CommonPasswordValidator(DjangoCommonPasswordValidator):
```
Comment (herzog0) [inline users/validators.py:7]: Is this validator being used anywhere else except from test cases? I couldn't find any other uses

**Review by herzog0 — COMMENTED** [review]
Heya, looks good to me, just a couple of things I didn't quite understand if you could comment on them, thanks!

File: users/views.py (hunk 3 above)
Comment (ycanales) [reply users/views.py]: it is, once #2412 get merged we can go back to v3-login, or, since it's close to being merged I could remove this and just let Katty know that after resetting the password, the link will point to v3 login which is not ready yet in this branch.

Or I could rebase over #2412? so many options haha what do you think?

File: users/validators.py (hunk 4 above)
Comment (ycanales) [reply users/validators.py:7]: it is! in the allauth settings 
https://github.com/boostorg/website-v2/pull/2488/changes#diff-935426a2f59d34e9506987404f8d78b5b5b43431694101f5be37ad65af4cf193L240

**Comment by ycanales** [comment]
Updated with the last changes that include the new Sign In which is now wired into the call to action button after resetting the password. Tested OK. Moving to Ready for QA.

**Review by kattyode — APPROVED** [review]
QA Approved
